### PR TITLE
feat(dashboard): Add test inbox for full E2E test journey

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* Add test inbox for full E2E test journey with in-app channel
  * The test inbox displays when visiting the `/test` route, demonstrating the end-user `<Inbox />` experience directly from the test page. Both the real-time delivery inbox and the preferences page are accessible, allowing the user to preview the entire inbox experience as the subscriber would see it
  * The normal inbox displays when not visiting the `/test` route, ensuring Novu platform messages can be delivered to the user
* Add 50% more oscillation to bell animation
* Add padding to `@novu/js` empty state component

### Screenshots
_Updated - Inbox with test visual cues_

image

https://github.com/user-attachments/assets/2e4bdbd1-b177-4dec-bbeb-a77ce449a75d

_Demonstration of E2E testing experience from test page_

https://github.com/user-attachments/assets/431474f0-ccce-4416-afd0-503effdfc0d7

_Test inbox default page_
image

_Test inbox unread page_
image

_Test inbox archived page_
image

_Test inbox preferences page_
image

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/js/src/ui/components/Notification/NotificationList.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const EmptyNotificationList = () => {
     <div
       class={style(
         'notificationListEmptyNoticeContainer',
-        'nt-absolute nt-inset-0 nt-flex nt-flex-col nt-items-center nt-m-auto nt-h-fit nt-w-full nt-text-foreground-alpha-100'
+        'nt-flex nt-flex-col nt-items-center nt-m-auto nt-h-fit nt-w-full nt-text-foreground-alpha-100 p-6 text-center'
```
Comment (rifont) [inline packages/js/src/ui/components/Notification/NotificationList.tsx]: I tried adding the `nt-` prefix but I couldn't seem to get the effects to apply. Is there an extra css postbuild step that is required to make the change propagate here?
```suggestion
        'nt-flex nt-flex-col nt-items-center nt-m-auto nt-h-fit nt-w-full nt-text-foreground-alpha-100 nt-p-6 nt-text-center'
```

File: apps/dashboard/src/components/icons/inbox-bell-filled.tsx [hunk 2]
```diff
@@ -3,16 +3,23 @@ import type { HTMLAttributes } from 'react';
...
 
 export function InboxBellFilled(props: HTMLAttributes<HTMLOrSVGElement> & InboxBellFilledProps) {
   return (
     <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="none" viewBox="0 0 9 12" {...props}>
-      <path
-        className={props.bellClassName}
-        fill="currentColor"
-        d="M4.5.856a.642.642 0 0 0-.643.643v.386a3.216 3.216 0 0 0-2.572 3.15v.377c0 .945-.347 1.857-.974 2.564l-.149.167a.642.642 0 0 0 .48 1.07h7.715a.644.644 0 0 0 .48-1.07l-.149-.167a3.863 3.863 0 0 1-.974-2.564v-.377a3.216 3.216 0 0 0-2.572-3.15v-.386A.642.642 0 0 0 4.5.856Z"
-      ></path>
+      <g className={props.bellClassName}>
```
Comment (rifont) [inline apps/dashboard/src/components/icons/inbox-bell-filled.tsx:12]: Wrapping the code indicator in the group to have it oscillate with the bell.

File: apps/dashboard/tailwind.config.js [hunk 3]
```diff
@@ -153,15 +153,15 @@ export default {
...
-          '90%': { transform: 'rotate(0.2deg)' },
+          '10%': { transform: 'rotate(3deg)' },
+          '20%': { transform: 'rotate(-3deg)' },
+          '30%': { transform: 'rotate(2.25deg)' },
+          '40%': { transform: 'rotate(-2.25deg)' },
+          '50%': { transform: 'rotate(1.5deg)' },
+          '60%': { transform: 'rotate(-1.5deg)' },
+          '70%': { transform: 'rotate(0.75deg)' },
+          '80%': { transform: 'rotate(-0.75deg)' },
+          '90%': { transform: 'rotate(0.3deg)' },
```
Comment (rifont) [inline apps/dashboard/tailwind.config.js:164]: Making the jingle animation a bit stronger, it was difficult to see earlier. It's still subtle enough.

File: apps/dashboard/tailwind.config.js [hunk 4]
```diff
@@ -182,6 +182,9 @@ export default {
         swing: 'swing 3s ease-in-out',
         jingle: 'jingle 3s ease-in-out',
       },
+      backgroundImage: {
+        'test-pattern': 'repeating-linear-gradient(135deg, hsl(var(--neutral-100)) 0, hsl(var(--neutral-100)) 2px, hsl(var(--neutral-200)) 2px, hsl(var(--neutral-200)) 4px)',
+      },
```
Comment (rifont) [inline apps/dashboard/tailwind.config.js:187]: Reusable separated-lined pattern for test backgrounds.

File: packages/js/src/ui/components/Notification/NotificationList.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ const EmptyNotificationList = () => {
     <div
       class={style(
         'notificationListEmptyNoticeContainer',
-        'nt-absolute nt-inset-0 nt-flex nt-flex-col nt-items-center nt-m-auto nt-h-fit nt-w-full nt-text-foreground-alpha-100'
+        'nt-flex nt-flex-col nt-items-center nt-m-auto nt-h-fit nt-w-full nt-text-foreground-alpha-100 nt-p-6 nt-text-center'
```
Comment (rifont) [inline packages/js/src/ui/components/Notification/NotificationList.tsx:20]: Adding padding to the empty state, to ensure custom localization prop has appropriate padding. The added padding is equivalent to the padding around other parts of the view (p-6)

image

File: apps/dashboard/src/components/header-navigation/header-button.tsx [hunk 6]
```diff
@@ -1,21 +1,27 @@
...
   return (
     <Tooltip>
       <TooltipTrigger asChild>
         <div
           tabIndex={0}
-          className="hover:bg-foreground-100 focus-visible:ring-ring flex h-6 w-6 cursor-pointer items-center justify-center rounded-2xl transition-[background-color,box-shadow] duration-200 ease-in-out focus-visible:outline-none focus-visible:ring-2"
+          className={cn(
+            `hover:bg-foreground-100 focus-visible:ring-ring flex h-6 w-6 cursor-pointer items-center justify-center rounded-2xl transition-[background-color,box-shadow] duration-200 ease-in-out focus-visible:outline-none focus-visible:ring-2`,
+            className
+          )}
```
Comment (rifont) [inline apps/dashboard/src/components/header-navigation/header-button.tsx:24]: Enabling the `bg-test-pattern` class to be passed through

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**